tuxemon/core/states/dialog/__init__.py

```diff
@@ -21,13 +22,15 @@ class DialogState(PopUpMenu):
     def startup(self, **kwargs):
         super(DialogState, self).startup(**kwargs)
         self.text_queue = kwargs.get("text", list())
+        self.avatar = kwargs.get("avatar", None)
         self.menu = kwargs.get("menu", None)
         self.text_area = TextArea(self.font, self.font_color)
         self.text_area.rect = self.calc_internal_rect()
         self.sprites.add(self.text_area)
 
     def on_open(self):
         self.next_text()
+        self.draw_avatar()
```

Comment:
please remove this and see another comment about drawing.  drawing should only happen inside the "draw" method.

Reply:
Latest commit refractors this: The function to pick the avatar was moved to tools.py and only used once when initializing the dialog. The dialog state is now called with the sprite that was fetched, and is only responsible for adding it on startup the same way it adds the text area.